llvm/include/llvm/IR/Attributes.td

```diff
@@ -359,7 +366,10 @@ def : CompatRule<"isEqual<ShadowCallStackAttr>">;
 def : CompatRule<"isEqual<UseSampleProfileAttr>">;
 def : CompatRule<"isEqual<NoProfileAttr>">;
 def : CompatRule<"checkDenormMode">;
-
+def : CompatRule<"checkStrictFP">;
+def : CompatRuleAttr<"isEqual", "sign-return-address">;
+def : CompatRuleAttr<"isEqual", "sign-return-address-key">;
+def : CompatRuleAttr<"isEqual", "branch-protection-pauth-lr">;
```

Comment:
I guess I'm surprised that you can't do:
```
def : CompatRule<"isEqual<sign-return-address>">;
```
Is this because these attributes are string attributes?  If so, consider adding more to the comment blocks to describe //when// a developer should use `CompatRule` vs `CompatRuleAttr`.  Perhaps s/CompatRuleAttr/CompatRuleStrAttr/ since the existing vanillla CompatRule is already concerned with function attributes in order to better distinguish that this new class is for //string// attributes.

Reply:
> Is this because these attributes are string attributes?
Correct, and string can't be a template param ( `isEqual<"sign-return-address">"` won't work)